Tests/RegexTests/MatchTests.swift

```diff
@@ -1117,6 +1121,43 @@ extension RegexTests {
     firstMatchTest(#"(?s:.+)"#, input: "a\nb", match: "a\nb")
   }
   
+  func testCaseSensitivity() {
+    matchTest(
+      #"c..e"#,
+      ("cafe", true),
+      ("Cafe", false))
+    matchTest(
+      #"(?i)c.f."#,
+      ("cafe", true),
+      ("Cafe", true),
+      ("caFe", true))
+    matchTest(
+      #"(?i)cafe"#,
+      ("cafe", true),
+      ("Cafe", true),
+      ("caFe", true))
+    matchTest(
+      #"(?i)café"#,
+      ("café", true),
+      ("CafÉ", true))
+    matchTest(
+      #"(?i)\u{63}af\u{e9}"#,
+      ("café", true),
+      ("CafÉ", true))
+    
+    matchTest(
+      #"[caFE]{4}"#,
+      ("cafe", false),
+      ("CAFE", false),
+      ("caFE", true),
+      ("EFac", true))
+    matchTest(
+      #"(?i)[caFE]{4}"#,
+      ("cafe", true),
+      ("CaFe", true),
+      ("EfAc", true))
+  }
```

Comment:
Thanks for the tests! What about multi-scalar expansions, like the German SS?

Reply:
I believe we need case folding to do all this properly, don't we? I'll follow up with some tests showing the holes.